packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -325,69 +325,48 @@ const numberWithUnitCellView = NumberWithUnitCellView.compose({
 
 #### Exposing cell view events externally
 
-<!--
-TODO: Being actively discussed in: https://github.com/ni/nimble/pull/1153#discussion_r1160996644
-
-Finally, here is a column element that allows a user to register a callback for a click event on a button inside the cell template:
+Because the cell view is not a descendant of the column element, we must add special support for the column to handle events that originate in the cells. A column will define a `columnDelegatedEvents` property which is an array of event names that should be delegated. As this is a static property, it will be part of the `ColumnInternalsOptions` passed to the base column constructor.
 
 ```TS
-type TableColumnButtonCellData = TableStringField<'id'>;
-
-public class TableColumnButton extends TableColumn<TableColumnButtonCellData> {
+export interface ColumnInternalsOptions {
     ...
+    readonly columnDelegatedEvents: readonly string[];
+    ...
+}
 
-    public cellRecordFieldNames = ['id'] as const;
-
-    @attr
-    public idKey: string;
-
-    public getDataRecordFieldNames(): string[] {
-        return [valueKey];
-    }
-
-    public cellViewTag = 'table-cell-view-button';
-
-    public validateCellData(cellData: TCellRecord): void {
-        if (typeof(cellData['id']) !== 'string') {
-            throw new Error('Type for cellData is incorrect!');
-        }
+export class ColumnInternals<TColumnConfig> {
+    ...
+    public readonly columnDelegatedEvents: readonly string[];
+    ...
+    public constructor(options: ColumnInternalsOptions) {
+        this.columnDelegatedEvents = options.columnDelegatedEvents;
+        ...
     }
 }
 
-class ButtonCellView extends TableCellView<TableColumnButtonCellData> {
-    public onButtonClick(): void {
-        this.$emit('button-click', { data: this.cellRecord.id })
+AnchorTableColumn extends TableColumn {
+    constructor() {
+        super({
+            columnDelegatedEvents: ['click'],
+            ...
+        });
     }
+    ...
 }
-const buttonCellView = ButtonCellView.compose({
-    baseName: 'table-cell-view-button',
-    template: html<ButtonCellView>`
-        <nimble-button @click="${(x) => x.onButtonClick()}">
-            <span>Press Me</span>
-        </nimble-button>`,
-    styles: /* styling */
-});
 ```
 
-Angular template:
-
-```HTML
-<nimble-table (button-click)="doSomething($event)">
-    <nimble-table-column-button></nimble-table-column-button>
-</nimble-table>
-```
-
-Angular code:
+Upon connecting a cell view to the DOM, we will attach an event listener for each event type in `columnDelegatedEvents`. The handler will wrap the original event in a new `CustomEvent` and dispatch that to the column. It will also stop propagation on the original event, and prevent default if the `dispatchEvent` call returns `false`.
 
 ```TS
-private doSomething(CustomEvent e): void {
-    ...
-}
-
+this.addEventListener(delegatedEventName, event => {
+    event.stopPropagation();
```

Comment:
I'm on the fence about this. If anything it just feels inconsistent with other events that fire from table cells. I don't think there is a real concern about getting them confused as the names are different, i.e. `delegated-event` vs `click`. I'd lean to letting the column author decide instead if they want.

The rest of the layers of the table don't care about that event propagating I think. So not sure why we would force that behavior.

Reply:
Not sure I follow what you're saying about getting event names confused. The reason I suggested we stop propagation is that it matched my mental model of us "teleporting" the event to the column element. I figured if we were re-routing it, it should no longer continue along the original bubble-up path. However, I concede that's not exactly what we're doing, as we're dispatching a new event, not the original one. I've removed the call to `stopPropagation()`. I guess as long as other handlers up the tree (if any) honor the `defaultPrevented` flag, there's no harm in letting it propagate.

Reply:
> Not sure I follow what you're saying about getting event names confused

That was a guess at trying to understand why the default was prevented.

> we're dispatching a new event, not the original one

👍 yea that's the thought process I was going down. Sounds good